crates/locutus-runtime/src/versioning.rs

```diff
@@ -0,0 +1,167 @@
+use std::fmt;
+use std::fmt::{Display, Formatter};
+use std::fs::File;
+use std::io::{Cursor, Read};
+use std::path::Path;
+use std::sync::Arc;
+
+use byteorder::{BigEndian, ReadBytesExt};
+use locutus_stdlib::contract_interface::ContractKey;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+use crate::WrappedContract;
+use locutus_stdlib::prelude::{ContractCode, Parameters, TryFromTsStd, WsApiError};
+
+/// Wrapper that allows contract versioning. This enum maintains the types of contracts that are
+/// allowed and their corresponding version.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum ContractContainer {
+    Wasm(WasmAPIVersion),
+}
+
+impl ContractContainer {
+    /// Return the `ContractContainer` content from the specified path as as `Vec<u8>`.
+    pub(crate) fn get_contract_data_from_fs(path: &Path) -> Result<Vec<u8>, std::io::Error> {
+        let mut contract_file = File::open(path)?;
+        let mut contract_data = if let Ok(md) = contract_file.metadata() {
+            Vec::with_capacity(md.len() as usize)
+        } else {
+            Vec::new()
+        };
+        contract_file.read_to_end(&mut contract_data)?;
+        Ok(contract_data)
+    }
+
+    /// Return the `ContractKey` from the specific contract version.
+    pub fn get_key(&self) -> ContractKey {
```

Comment:
instead of prefixing the methods with `get_*` let's just call them directly as the fields they are getting (eg. `key(), params()` etc.). Distinguising between getting and setting in Rust is not very idiomatic since the language is already letting you know through &mut or & if something is mutable or not.

Reply:
Done